components/rank/index.tsx

```diff
@@ -0,0 +1,171 @@
+import styled from 'styled-components';
+import byteSize from 'byte-size';
+
+import { ArrowDown, ArrowUp } from '@components/icons';
+
+import device from '@utils/media_queries';
+import type { Data } from '@utils/dataType';
+
+const RankWrapper = styled.div`
+	font-size: 0.75rem;
+
+	@media ${device.tablet} {
+		font-size: 1rem;
+
+		& th {
+			&:first-child {
+				padding-left: 2rem;
+			}
+
+			&:last-child {
+				padding-right: 2rem;
+			}
+		}
+
+		& td {
+			&:first-child > span {
+				padding-left: 2rem;
+			}
+
+			&:last-child > span {
+				padding-right: 2rem;
+			}
+		}
+	}
+
+	& > table {
+		width: 100%;
+		border-collapse: collapse;
+	}
+
+	& th {
+		text-align: left;
+		padding-top: 1.75rem;
+		padding-bottom: 1.75rem;
+		background: #121212;
+		position: sticky;
+		top: 0;
+
+		&:first-child {
+			padding-left: 1rem;
+		}
+
+		&:last-child {
+			padding-right: 1rem;
+		}
+	}
+
+	& td {
+		&:first-child > span {
+			border-radius: 8px 0 0 8px;
+			padding-left: 1rem;
+		}
+
+		&:last-child > span {
+			border-radius: 0 8px 8px 0;
+			padding-right: 1rem;
+		}
+
+		& > span {
+			display: block;
+			background: #2c2c2c;
+			padding-top: 1rem;
+			padding-bottom: 1rem;
+			margin-bottom: 1rem;
+		}
+	}
+`;
+
+const formatWalletAddress = (wallet: string): string => {
+	const first5Chars = wallet.slice(0, 5);
+	const last5Chars = wallet.slice(-5);
+
+	return `${first5Chars}...${last5Chars}`;
+};
+
+const positionIndicator = (position: number): string => {
+	const indicator = Math.abs(position);
+	const cent = indicator % 100;
+	if (cent >= 10 && cent <= 20) return 'th';
+	const dec = indicator % 10;
+	if (dec === 1) return 'st';
+	if (dec === 2) return 'nd';
+	if (dec === 3) return 'rd';
+	return 'th';
+};
+
+const calculateByteSize = (bytes: number): string => {
+	const size = byteSize(bytes, { units: 'iec' });
```

Comment:
What exactly is this package doing here? Are we converting from decimal/binary or something? Arweave and the ArDrive apps only handle bytes in binary

We do have a `formatBytes` function in Core that performs something similar. We could just copy paste a version of that function into a utils file rather than having a whole dependency added here.


```ts
// Format byte size to something nicer.  This is minified...
export function formatBytes(bytes: number): string {
	const marker = 1024; // Change to 1000 if required
	const decimal = 3; // Change as required
	const kiloBytes = marker; // One Kilobyte is 1024 bytes
	const megaBytes = marker * marker; // One MB is 1024 KB
	const gigaBytes = marker * marker * marker; // One GB is 1024 MB
	// const teraBytes = marker * marker * marker * marker; // One TB is 1024 GB

	// return bytes if less than a KB
	if (bytes < kiloBytes) return `${bytes} Bytes`;
	// return KB if less than a MB
	if (bytes < megaBytes) return `${(bytes / kiloBytes).toFixed(decimal)} KB`;
	// return MB if less than a GB
	if (bytes < gigaBytes) return `${(bytes / megaBytes).toFixed(decimal)} MB`;
	// return GB if less than a TB
	return `${(bytes / gigaBytes).toFixed(decimal)} GB`;
}
```